sklearn/inspection/_partial_dependence.py

```diff
@@ -88,17 +89,26 @@ def _grid_from_X(X, percentiles, is_categorical, grid_resolution):
 
     values = []
     for feature, is_cat in enumerate(is_categorical):
-        uniques = np.unique(_safe_indexing(X, feature, axis=1))
+        uniques = _unique(_safe_indexing(X, feature, axis=1))
```

Comment:
For my education: why is the change from `np.unique` to `_unique` needed? When I call `np.unique(X[:, 0])` (with `X` setup like in the non-regression test) I get a list of values with a `nan` as the last element. This is the same as I get when I use `_unique`.

It seems like the important change is the next line which filters out `nan` values.

Reply:
`np.unique` will try to do a sorting and from the issue, you can get an error:

```python
TypeError: '<' not supported between instances of 'str' and 'float'
```

I assume this happens when dealing with `str` and `np.nan`. `_unique` was written for the encoders and handles `nan` values.